### PR TITLE
CASSANDRA-15756 in-jvm dtest IInstance and ICoordinator should use QueryResult as the base API

### PR DESCRIPTION
**Pull request by dcapwell**

### CODE REVIEW

File: src/main/java/org/apache/cassandra/distributed/api/CompleteQueryResult.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
+/*
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/api/CompleteQueryResult.java:1]: this isn't a new file but renamed `QueryResult` to this and made `QueryResult` a interface.

**Comment by ifesdjeen** [comment]
I've pushed several suggestions here (thought might be quicker this way): https://github.com/apache/cassandra-in-jvm-dtest-api/compare/master...ifesdjeen:feature/allApiUseQueryResult?expand=1

Caan be summarised as:
  * should we move _classes_ to `shared` package and leave only interfaces in `api`?
  * remove `get` from some methods
  * rename `Complete` to `SImple` result

I think these API changes will also help us with paged query coordinator switch in the future!

**Comment by dcapwell** [comment]
> should we move classes to shared package and leave only interfaces in api?

I don't agree with that if the class is part of the API.  The main one I see in your commit is `Row` which to me is part of the API.  Anything a interface returns is part of the API to me.

**Comment by dcapwell** [comment]
@ifesdjeen https://github.com/apache/cassandra-in-jvm-dtest-api/pull/10/commits/ca592b43531c7ce1f5df1d252c78de63b456c952. this is based off your feedback.  the only real diff is I didn't move anything to shared.

**Pull request closed without merging**